Firebase/Core/Private/FIRAppInternal.h

```diff
@@ -174,12 +174,14 @@ typedef NSString *_Nullable (^FIRAppGetUIDImplementation)(void);
  * Registers a given third-party library with the given version number to be reported for
  * analyitcs.
  *
- * @param library Name of the library
+ * @param library Name of the library. It's typically the CocoaPod name
  * @param version Version of the library
  */
 // clang-format off
 + (void)registerLibrary:(NSString *)library
-            withVersion:(NSString *)version NS_SWIFT_NAME(registerLibrary(_:version:));
```

Comment:
I was thinking we could change one of the protocols (or introduce a new one) so the call-site is much cleaner:

```
@protocol FIRLibrary <NSObject>
@required
+ (NSString *)libraryName;
+ (NSString *)libraryVersion;

@optional
+ (NSArray<FIRComponent *> *)componentsToRegister;
+ (void)configureWithApp:(FIRApp *)app;

@end

// Then provide a call in `FIRAppInternal.h`:
+ (void)registerLibrary:(id<FIRLibrary>)library;
```

That way there's only one protocol that defines a library and the two optional methods can be totally ignored if requested. Thoughts?